### ISSUE TITLE
Add a way to customize the input

### ISSUE THREAD

**Issue by olzraiti** [furqanZafar/react-selectize#124]
Currently it's impossible to customize the text input that the component wraps. Customizing the text input would allow users to change the `type` attribute or set the input's id (currently you can't add an id the the react-selectize component and therefore you can't focus the component when a label is clicked). I can think of two ways how the text input could be customized:

1. Add some parameter object like `inputOptions` which would be passed to the React input element.
2. Add a input rendering function parameter.

Adding a way to customize the input would solve at least #79 #118.

**Comment by green2k** [comment]
Yeah, it would be also perfect if we're able to set the `readOnly` property! 

https://github.com/furqanZafar/react-selectize/issues/172
